src/lib/JIT.hs

```diff
@@ -304,19 +304,23 @@ compileInstr instr = case instr of
   ICastOp idt ix -> (:[]) <$> do
     x <- compileExpr ix
     let (xt, dt) = (L.typeOf x, scalarTy idt)
-    case (xt, dt) of
-      (L.IntegerType _, L.IntegerType _) -> x `asIntWidth` dt
-      (L.FloatingPointType fpt, L.FloatingPointType fpt') -> case compare fpt fpt' of
-        LT -> emitInstr dt $ L.FPExt x dt []
-        EQ -> return x
-        GT -> emitInstr dt $ L.FPTrunc x dt []
-      (L.FloatingPointType _, L.IntegerType _) -> emitInstr dt $ L.FPToSI x dt []
-      (L.IntegerType _, L.FloatingPointType _) -> emitInstr dt $ L.SIToFP x dt []
-      (L.PointerType _ _, L.PointerType eltTy _) -> castLPtr eltTy x
-      (L.IntegerType 64 , ptrTy@(L.PointerType _ _)) ->
-        emitInstr ptrTy $ L.IntToPtr x ptrTy []
-      (L.PointerType _ _, L.IntegerType 64) -> emitInstr i64 $ L.PtrToInt x i64 []
-      _ -> error $ "Unsupported cast"
+    case (xt, idt) of
+      -- if upcasting to unsigned int, use zext instruction
+      (L.IntegerType _, Scalar Word64Type) -> x `zeroExtendTo` dt
+      (L.IntegerType 8, Scalar Word32Type) -> x `zeroExtendTo` dt
```

Comment:
this cast (W8ToW32) is actually not exposed in Dex, but I figured it would be good to add here defensively?

Reply:
Sure. Why specialize it to Word8 though? Seems fine to put an underscore in there

Reply:
How about writing it out as
```hs
(L.IntegerType bits, Scalar Word32Type) | bits < 32 -> x `zeroExtendTo` dt
```

Reply:
that seems much nicer! thanks